csharp/ql/src/semmle/code/csharp/dataflow/TaintTracking.qll

```diff
@@ -99,6 +104,258 @@ module TaintTracking {
     }
   }
 
+  /**
+   * A taint tracking configuration.
+   *
+   * A taint tracking configuration is a special dataflow configuration
+   * (`DataFlow::Configuration`) that allows for flow through nodes that do not
+   * necessarily preserve values, but are still relevant from a taint tracking
+   * perspective. (For example, string concatenation, where one of the operands
+   * is tainted.)
+   *
+   * Each use of the taint tracking library must define its own unique extension
+   * of this abstract class. A configuration defines a set of relevant sources
+   * (`isSource`) and sinks (`isSink`), and may additionally treat intermediate
+   * nodes as "sanitizers" (`isSanitizer`) as well as add custom taint flow steps
+   * (`isAdditionalTaintStep()`).
+   */
+  abstract class Configuration2 extends DataFlow2::Configuration {
```

Comment:
This is inconsistent with the way DataFlow is duplicated. Would it make sense to
- Split up `TaintTracking.qll` into `TaintTrackingConfigurationN.qll`
- Name the configurations `TaintTrackingN::Configuration` instead of `TaintTracking::ConfigurationN`
- It's not clear that these are close copies of each other. How will they be kept in sync? Maybe better to make use of `identical-files.json` here as well.

Reply:
I followed the pattern from Java, but I agree that it is inconsistent. Any thoughts on why you did it this way for Java, @aschackmull ?

Reply:
I agree that `TaintTrackingN::Configuration` instead of `TaintTracking::ConfigurationN` would be more consistent - there's no real reason why it ended up with this inconsistency for Java.  Unfortunately we cannot use the `identical-files.json` mechanism to keep this in sync, as the copied code is just below class level.